src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -718,6 +718,14 @@ emit_hardware_intrinsics (
 		if (!info)
 			goto support_probe_complete;
 		id = info->id;
+
+#ifdef TARGET_ARM64
+		if (!(cfg->compile_aot && cfg->full_aot)) {
```

Comment:
I think this also needs to check `cfg->interp`:
```suggestion
		if (!(cfg->compile_aot && cfg->full_aot && !cfg->interp)) {
```

Also, not sure if we should enable this for llvmonly, too.  @vargaz what do you think?